docs/DeveloperGuide.md

```diff
@@ -1,34 +1,148 @@
 # Developer Guide
+* [Setting up](#1-setting-up)
+* [Design](#2-design)
+* [Implementation](#3-implementation)
+  * [Undo features](#31-proposed-undo-feature)
+    * [View By Category](#311-view-by-category)
+* [Documentation](#4-documentation)
+* [Testing](#5-testing)
+* [Dev Ops](#6-dev-ops)
+  * [Appendix A: Product Scope](#61-appendix-aproduct-scope)
+  * [Appendix B:Value proposition](#62-appendix-bvalue-proposition)
+  * [Appendix C:User Stories](#63-appendix-cuser-stories)
+  * [Appendix D:Non-Functional Requirements](#64-appendix-dnon-functional-requirements)
+  * [Appendix E:Glossary](#65-appendix-eglossary)
+  * [Appendix F:Instructions for Manual Testing](#66-appendix-finstructions-for-manual-testing)
+## 1. Setting up
+## 2. Design
 
-## Design & Implementation
+{Describe the design of the product. Use **Architecture Diagram** which has not been covered yet.}
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+It extends the current view command.
+The user can use `list` to view all tasks.
+The user can also use `list CATEGORY` to view specific classes whose category is sth. 
+This is the UML design for view by category.
+![UML for View](images/viewCategory.png)
 
+<p>
+Step1: Users add task and add either a default category/new category.
+If user adds default category, then the categoryMap in taskList will not be modified.
+If the user adds a new category, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
+</p>
 
-## Product Scope
-### Target user profile
 
-{Describe the target user profile}
+Step2:  User view tasks by category.
+If user only type in `list`, then all tasks will be displayed according to the sequence of task
+index. It simply calls the view function.
+If user type in in `list CATEGORY`, and the category is valid, then it will go for the
+returnCategory method.
+If user type in in `list CATEGORY`, and the category is invalid, then it will go for the
+displayCategory method. 
+Whatever the path, the UI will finally be called either display tasks given the index or display
+category. 
```

Comment:
The formatting of this portion makes it a little difficult for me to read. Would it be better to use bullet points to separate the various `if` cases?